### PR TITLE
Use the smart_proxy_plugin.yml reusable GHA workflow

### PR DESCRIPTION
**Pull request by ekohl**
This is largely the same as test-gem, except it verifies that it at least implements support for Ruby versions that the Smart Proxy wants. It also implements the version selector by respecting SMART_PROXY_BRANCH in the Gemfile.

Also cleans up Jenkins bits and updates the license tag.

### CODE REVIEW

**Comment by timdeluxe** [comment]
@ekohl I think you see the failing tests here as well, in this case it looks like problems with the unit tests?

**Comment by ekohl** [comment]
That's the result of [Drop Jenkins integration in tests](https://github.com/theforeman/smart_proxy_dns_infoblox/pull/34/commits/810456568626572384f3a40b1a0c4a726d973114). I could leave it out until we merge https://github.com/theforeman/jenkins-jobs/pull/506.

**Comment by timdeluxe** [comment]
Well you are the specialist, i am fine with it, i leave it up to you. We could also merge despite the failing tests. Or did you mean exactly that?

**Comment by ekohl** [comment]
I'll wait a bit to see about the other repos. I'm leaning to merging them all at the same time and then clean up Jenkins. Then we can merge this while ignoring the failing tests because the check will disappear on the next PR.

**Review by evgeni — APPROVED** [review]

**Pull request merged**
